Error when creating test data

Dear Rems team,

we are testing this latest release (nkr-20191125) and we are getting his error while generating test data set

```
$ lein run test-data
2019-11-26 00:03:57,035 [main - - -] INFO  org.eclipse.jetty.util.log - Logging initialized @4184ms to org.eclipse.jetty.util.log.Slf4jLog
2019-11-26 00:04:20,713 [main - - -] INFO  com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Starting...
2019-11-26 00:04:20,864 [main - - -] INFO  com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Start completed.
2019-11-26 00:04:20,870 [main - - -] INFO  rems.locales - Loaded extra translations from example-theme/extra-translations/en.edn
2019-11-26 00:04:20,876 [main - - -] WARN  rems.locales - Unused translation keys defined in example-theme/extra-translations/en.edn : ((:t :unused-key))
2019-11-26 00:04:20,885 [main - - -] WARN  rems.locales - Nonmatching format parameters in example-theme/extra-translations/en.edn : ({:key (:t :create-license :license-text), :translations #{}, :extra-translations #{"%1"}})
2019-11-26 00:04:20,887 [main - - -] INFO  rems.locales - Loaded extra translations from example-theme/extra-translations/fi.edn
2019-11-26 00:04:20,888 [main - - -] INFO  rems.standalone - Creating test data
Exception in thread "main" Syntax error compiling at (/tmp/form-init6707534227265926922.clj:1:73).
        at clojure.lang.Compiler.load(Compiler.java:7647)
        at clojure.lang.Compiler.loadFile(Compiler.java:7573)
        at clojure.main$load_script.invokeStatic(main.clj:452)
        at clojure.main$init_opt.invokeStatic(main.clj:454)
        at clojure.main$init_opt.invoke(main.clj:454)
        at clojure.main$initialize.invokeStatic(main.clj:485)
        at clojure.main$null_opt.invokeStatic(main.clj:519)
        at clojure.main$null_opt.invoke(main.clj:516)
        at clojure.main$main.invokeStatic(main.clj:598)
        at clojure.main$main.doInvoke(main.clj:561)
        at clojure.lang.RestFn.applyTo(RestFn.java:137)
        at clojure.lang.Var.applyTo(Var.java:705)
        at clojure.main.main(main.java:37)
Caused by: java.lang.Exception: Exception in :add-api-key!
        at conman.core$try_query$fn__13580$fn__13581.invoke(core.clj:32)
        at rems.db.core$eval15383$f__13608__auto____15410.invoke(core.clj:18)
        at rems.db.test_data$create_test_data_BANG_.invokeStatic(test_data.clj:814)
        at rems.db.test_data$create_test_data_BANG_.invoke(test_data.clj:813)
        at rems.standalone$_main.invokeStatic(standalone.clj:92)
        at rems.standalone$_main.doInvoke(standalone.clj:61)
        at clojure.lang.RestFn.invoke(RestFn.java:408)
        at clojure.lang.Var.invoke(Var.java:384)
        at user$eval140.invokeStatic(form-init6707534227265926922.clj:1)
        at user$eval140.invoke(form-init6707534227265926922.clj:1)
        at clojure.lang.Compiler.eval(Compiler.java:7176)
        at clojure.lang.Compiler.eval(Compiler.java:7166)
        at clojure.lang.Compiler.load(Compiler.java:7635)
        ... 12 more
Caused by: org.postgresql.util.PSQLException: ERROR: duplicate key value violates unique constraint "api_key_pkey"
  Detail: Key (apikey)=(42) already exists.
        at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2468)
        at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2211)
        at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:309)
        at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:446)
        at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:370)
        at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:149)
        at org.postgresql.jdbc.PgPreparedStatement.executeUpdate(PgPreparedStatement.java:124)
        at com.zaxxer.hikari.pool.ProxyPreparedStatement.executeUpdate(ProxyPreparedStatement.java:61)
        at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeUpdate(HikariProxyPreparedStatement.java)
        at clojure.java.jdbc$db_do_execute_prepared_return_keys$exec_and_return_keys__1523.invoke(jdbc.clj:927)
        at clojure.java.jdbc$db_do_execute_prepared_return_keys$fn__1532.invoke(jdbc.clj:954)
        at clojure.java.jdbc$db_transaction_STAR_.invokeStatic(jdbc.clj:771)
        at clojure.java.jdbc$db_transaction_STAR_.invoke(jdbc.clj:741)
        at clojure.java.jdbc$db_transaction_STAR_.invokeStatic(jdbc.clj:754)
        at clojure.java.jdbc$db_transaction_STAR_.invoke(jdbc.clj:741)
        at clojure.java.jdbc$db_do_execute_prepared_return_keys.invokeStatic(jdbc.clj:953)
        at clojure.java.jdbc$db_do_execute_prepared_return_keys.invoke(jdbc.clj:911)
        at clojure.java.jdbc$db_do_prepared_return_keys.invokeStatic(jdbc.clj:986)
        at clojure.java.jdbc$db_do_prepared_return_keys.invoke(jdbc.clj:963)
        at clojure.java.jdbc$db_do_prepared_return_keys.invokeStatic(jdbc.clj:988)
        at clojure.java.jdbc$db_do_prepared_return_keys.invoke(jdbc.clj:963)
        at clojure.java.jdbc$db_do_prepared_return_keys.invokeStatic(jdbc.clj:971)
        at clojure.java.jdbc$db_do_prepared_return_keys.invoke(jdbc.clj:963)
        at hugsql.adapter.clojure_java_jdbc.HugsqlAdapterClojureJavaJdbc.execute(clojure_java_jdbc.clj:11)
        at hugsql.adapter$eval13195$fn__13196$G__13175__13201.invoke(adapter.clj:3)
        at hugsql.adapter$eval13195$fn__13196$G__13174__13207.invoke(adapter.clj:3)
        at clojure.lang.Var.invoke(Var.java:399)
        at hugsql.core$db_fn_STAR_$y__13515.doInvoke(core.clj:457)
        at clojure.lang.RestFn.invoke(RestFn.java:445)
        at hugsql.core$db_fn_STAR_$y__13515.invoke(core.clj:447)
        at conman.core$try_query$fn__13580$fn__13581.invoke(core.clj:30)
        ... 24 more
```

please advise

thank you

--- a/resources/sql/queries.sql
+++ b/resources/sql/queries.sql
@@ -477,7 +477,8 @@ VALUES (:payload::jsonb, :status);
 
 -- :name add-api-key! :insert
 INSERT INTO api_key (apiKey, comment)
-VALUES (:apikey, :comment);
+VALUES (:apikey, :comment)
+ON CONFLICT DO NOTHING;
 
 -- :name get-api-key :? :1
 SELECT apiKey FROM api_key

--- a/src/clj/rems/db/test_data.clj
+++ b/src/clj/rems/db/test_data.clj
@@ -810,7 +810,14 @@
                       :comment ""})))))
     (log/info "Performance test applications created")))
 
+(defn assert-no-existing-data! []
+  (assert (empty? (db/get-application-events {}))
+          "You have existing applications, refusing to continue. An empty database is needed.")
+  (assert (empty? (db/get-catalogue-items {}))
+          "You have existing catalogue items, refusing to continue. An empty database is needed."))
+
 (defn create-test-data! []
+  (assert-no-existing-data!)
   (db/add-api-key! {:apikey 42 :comment "test data"})
   (create-test-users-and-roles!)
   (let [res1 (create-resource! {:resource-ext-id "urn:nbn:fi:lb-201403262"
@@ -879,6 +886,7 @@
       (db/set-catalogue-item-endt! {:id dynamic-expired :end (time/now)}))))
 
 (defn create-demo-data! []
+  (assert-no-existing-data!)
   (let [[users user-data] (case (:authentication rems.config/env)
                             :oidc [+oidc-users+ +oidc-user-data+]
                             [+demo-users+ +demo-user-data+])]

